_posts/2020-11-09-pcr-primer-calculator-using-r.md

```diff
@@ -0,0 +1,298 @@
+
+## The Game Plan
+Like all good scientists and coders, we must first plan out how we want to develop this calculator before we start coding in R. Spending quality time to plan out and design the code will save us time and effort in the future. When planning out a script or a simple block of code, I like to use *pseudocode*. Below you can see the our pseudocode. As we venture into each section of the code, I will give more explanation to each step of our pseudocode and explain what that will look like in R.
+
+**Overview**  
+1. Adding libraries to the R script
+2. Import the PCR primer data  
+  - Use readr to access the primer data from a csv files  
+  - Defensive coding: check the file for primer data   
+    - If no primers are found in csv file, abort the script  
+3. Loop: Calculate the melting temperature of each primer  
+  - loop through each row of the csv file until we have processed all rows with sequence data
+    - ensure the primer name (in column A) does not have any spaces in the name. If so, replace it with a "_"
+    - capitalized all letters in the primer sequence (in column E) using stringr
+    - lowercase all letters in the restriction site sequences (in column D) using stringr
+    - capitalize all letters in the leading strand sequence using stringr
+    - store the complete PCR primer (leading strand, restriction site, and primer sequence) together to a variable
+    - store the complete PCR primer with "5'" and "3'" wrapped around the sequence to another variable
+    - count the number of A's, T's, G's, C's, etc. in the annealing portion of the sequence to individual variables using stringr
+    - if the sequence has ambiguous bases:
+      - [Yes] calculate the minimum and maximum melting temperature
+      - [No] calculate the melting temperature
+    - create variables to hold the gc-percentage and annealing primer length
+    - create a data frame using the variables we created
+4. Export the results of the calculator to a folder as a new csv file
+  - use readr to write the data frame to a csv file.
+    - save the file using the row number and primer name
+  - close the For Loop
+
+Lets take a look at the expected input file for our script.  
+![input file layout]({{ site.url }}/assets/pcr-primer-calculator-using-r-input-file.jpg)
+
+For each row with sequencing information, we want to process and output the results of PCR primer.  
+![output folder layout]({{ site.url }}/assets/pcr-primer-calculator-using-r-output-folder.jpg)
+
+Make a note about each column. reference the column inside the code. so that we can denote each code block and what field that code relates to.  
```

Comment:
Capitalize the beginning of the sentence. Change the period to a comma between "code" and "so"